subsys/bluetooth/services/ots/ots_oacp_internal.h

```diff
@@ -80,7 +88,7 @@ struct bt_gatt_ots_oacp_proc {
 			uint32_t offset;
 			uint32_t len;
 			uint8_t mode;
-		} write_params;
+		} __packed write_params;
```

Comment:
If they are all protocol definitions then they should probably all have __packed then?

Reply:
On a 32-bit processor the outcome is no different but you are correct, it should be packed to ensure correct behavior on all architecture types.

Reply:
Since we don't have endianness-specific integer types like Linux, `__packed` also acts as a reminder that the integer struct members are in some protocol endianness and need to be converted to/from host endianness.

Reply:
Actually, looking at the use of these structs in the code it looks like you're missing the endianness conversions.

Reply:
These structures are populated by the `oacp_command_decode` function which uses the `net_buf_simple_pull_le` family of functions to do the endianness conversion. The reason they are being annotated with `__packed` attribute is that the `oacp_command_len_verify` function validates the procedure's request length against the size of these structures.

An alternate approach would be to remove the `__packed` attribute and provide definitions for the size of each procedure. I presume using `__packed` could have a performance impact on some architectures, would you rather see the definition approach?

Reply:
Yes, please only use __packed for structs which represent protocol-format data. Since we don't have endianness-annotated integer types like Linux this is the only way we can maintain at least a little bit of clarity and predictability for what endianness certain variables have.